src/gui_common/SegmentedBar.cs

```diff
@@ -0,0 +1,195 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   A ProgressBar that is split up into IconProgressBars, data is stored in a dictionary
+/// </summary>
+public class SegmentedBar : Control
+{
+    public PackedScene IconProgressBarScene = GD.Load<PackedScene>("res://src/gui_common/IconProgressBar.tscn");
+    public string Type;
+    public float MaxValue;
+
+    public Vector2 Size
+    {
+        get
+        {
+            return new Vector2(RectSize.x, RectSize.y);
+        }
+        set
+        {
+            RectSize = new Vector2(value[0], value[1]);
+            RectMinSize = RectSize;
+        }
+    }
```

Comment:
I feel like this size control is unnecessary and should instead be set in the godot editor.